resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -107,99 +107,80 @@ private[spark] class Client(
 
     val k8ClientConfig = k8ConfBuilder.build
     Utils.tryWithResource(new DefaultKubernetesClient(k8ClientConfig)) { kubernetesClient =>
+      val kubernetesComponentCleaner = new KubernetesComponentCleaner(kubernetesClient)
```

Comment:
there's a lot of passing around of this local variable.  I wonder if it makes more sense to make this instance local, add to it throughout the run and other methods and then at the end of the run method (probably in a finally block) clean everything up.  That approach reminds me a lot of a Haskell Monad accumulating state for later processing